### ISSUE TITLE
short-paths does not appear to be in use with code lens

### ISSUE THREAD

**Issue by yminsky** [ocamllabs/vscode-ocaml-platform#225]
In particular, the inferred types that are thrown back by hovering over an identifier are sensibly shortened, but the ones that show up via codelens are long and annoying!

![image](https://user-images.githubusercontent.com/715302/82734005-e9ab8f80-9ce5-11ea-88dd-cdd54b7f5784.png)

hovering over `fib` shows the expected type of `int -> int`.

**Comment by kyoDralliam** [comment]
Duplicate of #341

**Comment by rgrinberg** [comment]
This is fixed in the latest ocaml-lsp.

**Comment by yminsky** [comment]
Is this in opam yet? The inference still looks pretty busted to me. I'm on version 1.1.0.

![image](https://user-images.githubusercontent.com/715302/96945498-ee9a8800-14ab-11eb-86f9-2780037e73bd.png)

**Comment by rgrinberg** [comment]
I see. At least we fixed the mangled identifiers from showing up. @voodoos any idea why the identifiers are still needlessly qualified?

**Comment by yminsky** [comment]
It's also unpredictable.  If I change the code a bit, I can get it to say Base.Map.t instead of Map_intf.Map.t.

**Comment by voodoos** [comment]
I am able to reproduce the issue in vscode with latest ocaml-lsp (and external configuration readers).
However manually asking Merlin 3.4.1 for the outline of the file give the correct answer:

```
❯ ocamlmerlin single outline -position 3:19 -filename main.ml < main.ml | jq       
{
  "class": "return",
  "value": [
    {
      "start": {
        "line": 3,
        "col": 0
      },
      "end": {
        "line": 3,
        "col": 17
      },
      "name": "fib",
      "kind": "Value",
      "type": "int -> int",
      "children": [],
      "deprecated": false
    }
  ],
  "notifications": [],
  "timing": {
    "clock": 27,
    "cpu": 13,
    "query": 4,
    "pp": 0,
    "reader": 3,
    "ppx": 0,
    "typer": 6,
    "error": 0
  }
}
```

**Comment by ulugbekna** [comment]
Though breaks really seem unpredictable, a sure way to reproduce:

```ocaml
(* test.ml *)
open Core_kernel

let a = List.compare
```

1. Open `test.ml` (a real open, not preview open, i.e., filename in italics) and see short paths in codelens
2. Go to definition of `compare` fn in `Core_kernel`
3. Go back to `test.ml` and see that codelens paths are no longer short

**Comment by voodoos** [comment]
> Though breaks really seem unpredictable, a sure way to reproduce:
> 
> ```ocaml
> (* test.ml *)
> open Core_kernel
> 
> let a = List.compare
> ```
> 
>     1. Open `test.ml` (a real open, not preview open, i.e., filename in italics) and see short paths in codelens
> 
>     2. Go to definition of `compare` fn in `Core_kernel`
> 
>     3. Go back to `test.ml` and see that codelens paths are no longer short
> 
> 
> On edit or reopening the file, the codelens are fixed.

That my be related to the fact that files need to be edited after a build for the plugin to show updated information from the build artifacts ?

**Comment by rgrinberg** [comment]
This is fixed in lsp 1.6.1. short-paths is now enabled without reading the config from dune.
